org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.launching;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.IVMInstallType;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.launching.VMStandin;
+import org.eclipse.osgi.util.NLS;
+
+class DetectVMInstallationsJob extends Job {
+
+	private static final Object FAMILY = "Family " + DetectVMInstallationsJob.class.getName(); //$NON-NLS-1$
+	
+	DetectVMInstallationsJob() {
+		super(LaunchingMessages.lookupInstalledJVMs);
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
+		Collection<File> rootDirectories = new HashSet<>();
+		if (!Platform.OS_WIN32.equals(Platform.getOS())) {
+			rootDirectories.add(new File("/usr/lib/jvm")); //$NON-NLS-1$
+		}
+		rootDirectories.add(new File(System.getProperty("user.home"), ".sdkman/candidates/java")); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Shouldn't that also go under "non-Windows" if block? `.sdkman/candidates/java`

Reply:
I think SDKMan uses this directory on Windows as well.

Reply:
But surely not as `/candidates/java` ? Or will Java convert that path to `\candidates\java` on Windows automatically?

Reply:
I believe that either Java or Windows convert to `\` automatically (in `FileSystem.normalize()` I believe). If you have Windows, you can quickly test with `jshell`.
However, if you think it's important, I don't mind replacing `/` with `File.separator`

Reply:
https://stackoverflow.com/questions/2417485/difference-between-file-separator-and-slash-in-paths contains answers with references that claim `/` is OK to use in File on Windows as well.